mcs/class/corlib/System.IO/FileStream.cs

```diff
@@ -188,6 +188,12 @@ internal FileStream (string path, FileMode mode, FileAccess access, FileShare sh
 				throw new ArgumentException ("Name has invalid chars");
 			}
 
+			// IsolatedStorage needs to keep the Name property to the default "[Unknown]"
+			if (!anonymous) {
+				path = Path.InsecureGetFullPath (path);
```

Comment:
It does still use the `path` in all the following checks though so I'm not sure why we need different behavior.

To be clear what I'm asking is why is this not:

```csharp
path = Path.InsecureGetFullPath (path);

// IsolatedStorage needs to keep the Name property to the default "[Unknown]"
if (!anonymous)
    this.name = path;
```